src/state/nfts/createNftsStore.ts

```diff
@@ -0,0 +1,485 @@
+import { Address } from 'viem';
+import { arcClient } from '@/graphql';
+import { logger, RainbowError } from '@/logger';
+import { createQueryStore } from '@/state/internal/createQueryStore';
+import { time } from '@/utils';
+import { NftsState, NftParams, NftsQueryData, PaginationInfo, Collection } from './types';
+import { parseUniqueAsset, parseUniqueId } from '@/resources/nfts/utils';
+import { useBackendNetworksStore } from '../backendNetworks/backendNetworks';
+import { IS_DEV } from '@/env';
+import { queryClient } from '@/react-query';
+import { showcaseTokensQueryKey } from '@/hooks/useFetchShowcaseTokens';
+import { hiddenTokensQueryKey } from '@/hooks/useFetchHiddenTokens';
+
+export const PAGE_SIZE = 12;
+
+export const mergeMaps = (map1: Map<string, Collection>, map2: Map<string, Collection>) => {
+  return new Map(
+    (function* () {
```

Comment:
this is wild! very cool. any reason not to do `new Map([...map1, ...map2])` is it just faster?

Reply:
For large Maps, spreading iterates the content of both Maps twice. Once to create a temporary array containing the union of the two, then again to pass that into the Map constructor where it is iterated again to create the new Map. This generator function avoids all of that nonsense by not creating intermediate arrays.